test/java/com/odysseusinc/arachne/portal/front/StudyManagerTest.java

```diff
@@ -1,61 +1,699 @@
+
+        By titleBy = ByBuilder.modal(MODAL_TITLE_UPDATE_STUDY_TITLE);
+        waitFor(driver, titleBy);
+
+        WebElement modal = driver.findElement(titleBy);
+        WebElement studyTitleInput = modal.findElement(ByBuilder.input(PLACEHOLDER_STUDY_TITLE));
+        WebElement studySaveBtn = modal.findElement(ByBuilder.button("Save"));
+
+        String updatedPostfix = "dy";
+        studyTitleInput.sendKeys(updatedPostfix);
+        studySaveBtn.click();
+        waitFor(driver, ByBuilder.toolbar(STUDY_NAME));
+    }
+
+    @Test
+    public void test03UpdateStudyDescription() throws Exception {
+
+        loginAndOpenStudy();
+
+        By editBtn = ByBuilder.byClassAndText("ac-material-icons", "edit");
+        waitForPageLoad(driver, editBtn);
 
-        WebElement addStudyButton = driver.findElement(addBtn);
-        addStudyButton.click();
+        WebElement editButton = driver.findElement(editBtn);
+        editButton.click();
 
-        final By modal = ByBuilder.modal(MODAL_TITLE_CREATE_STUDY);
-        waitFor(driver, modal);
+        WebElement description = driver.findElement(ByBuilder.textArea("Add Study Objective"));
+        Assert.assertEquals("", description.getText());
 
-        WebElement modalWebElement = driver.findElement(modal);
-        WebElement studyNameInput = modalWebElement.findElement(ByBuilder.input(PLACEHOLDER_STUDY_TITLE));
-        WebElement studyTypeSelect = modalWebElement.findElement(ByBuilder.select(PLACEHOLDER_STUDY_TYPE));
-        WebElement studyTypeOption = modalWebElement.findElement(ByBuilder.selectOption(studyType));
-        WebElement studyCreateBtn = modalWebElement.findElement(ByBuilder.button("Create"));
+        String newDescription = "Long and beautiful description";
+        description.sendKeys(newDescription);
 
-        studyNameInput.sendKeys(studyName);
+        WebElement studyDescriptionBtn = driver
+                .findElement(By.className("ac-study-objective-edit"))
+                .findElement(ByBuilder.byClassAndText("ac-form__submit", "Save"));
+
+        studyDescriptionBtn.click();
+
+        waitFor(driver, ByBuilder.byClassAndText("ac-study-objective-view", newDescription));
+    }
+
+    @Test
+    public void test04UpdateStudyType() throws Exception {
+
+        loginAndOpenStudy();
+
+        String studyType = "Clinical Trial Design";
+
+        WebElement studyTypeSelect = driver.findElement(ByBuilder.select(studyType));
+        waitFor(driver, ByBuilder.select(studyType));
+
+        String updatedStudyType = "Health Economics and Outcomes";
+        WebElement studyTypeOption = driver.findElement(ByBuilder.selectOption(updatedStudyType));
 
         Actions actions = new Actions(driver);
         actions.moveToElement(studyTypeSelect).click().build().perform();
         actions.moveToElement(studyTypeOption).click().build().perform();
 
-        studyCreateBtn.click();
+        waitFor(driver, ByBuilder.select(updatedStudyType));
```

Comment:
Let's choose single naming: in study title edit you used BEFORE_TITLE and TITLE, here you are using TYPE and UPDATED_TYPE. (I would prefer second convention)

Reply:
there is only 1 case with BEFORE_TITLE and many with TITLE, so it's seems UPDATED_TITLE in this cases may be not understandable